ui-lib/src/main/java/co/electriccoin/zcash/ui/screen/integrations/viewmodel/IntegrationsViewModel.kt

```diff
@@ -0,0 +1,112 @@
+import kotlinx.coroutines.launch
+
+class IntegrationsViewModel(
+    getVersionInfo: GetVersionInfoProvider,
+    getZcashCurrency: GetZcashCurrencyProvider,
+    observeWalletState: ObserveWalletStateUseCase,
+    private val getTransparentAddress: GetTransparentAddressUseCase,
+    private val isFlexaAvailable: IsFlexaAvailableUseCase,
+    private val isCoinbaseAvailable: IsCoinbaseAvailableUseCase,
+) : ViewModel() {
+    val backNavigationCommand = MutableSharedFlow<Unit>()
+    val flexaNavigationCommand = MutableSharedFlow<Unit>()
+    val coinbaseNavigationCommand = MutableSharedFlow<String>()
+
+    private val versionInfo = getVersionInfo()
+    private val isDebug = versionInfo.let { it.isDebuggable && !it.isRunningUnderTestService }
+
+    private val isEnabled =
+        observeWalletState()
+            .map {
+                it != TopAppBarSubTitleState.Restoring
+            }
+
+    val state =
+        isEnabled.map { isEnabled ->
+            IntegrationsState(
+                version = stringRes(R.string.integrations_version, versionInfo.versionName),
+                coinbase =
+                    ZashiSettingsListItemState(
+                        // Set the wallet currency by app build is more future-proof, although we hide it from the UI
+                        // in the Testnet build
+                        text = stringRes(R.string.integrations_coinbase, getZcashCurrency.getLocalizedName()),
+                        subtitle =
+                            stringRes(
+                                R.string.integrations_coinbase_subtitle,
+                                getZcashCurrency.getLocalizedName()
+                            ),
+                        onClick = ::onBuyWithCoinbaseClicked
+                    ).takeIf { isCoinbaseAvailable() },
+                flexa =
+                    ZashiSettingsListItemState(
+                        text = stringRes(R.string.integrations_flexa),
+                        isEnabled = isEnabled,
+                        subtitle = stringRes(R.string.integrations_flexa_subtitle),
+                        onClick = ::onFlexaClicked
+                    ).takeIf { isFlexaAvailable() },
+                disabledInfo = stringRes(R.string.integrations_disabled_info).takeIf { isEnabled.not() },
+                onBack = ::onBack
+            )
+        }.stateIn(
+            scope = viewModelScope,
+            started = SharingStarted.WhileSubscribed(ANDROID_STATE_FLOW_TIMEOUT),
+            initialValue = null
+        )
+
+    fun onBack() =
+        viewModelScope.launch {
+            backNavigationCommand.emit(Unit)
+        }
+
+    private fun onBuyWithCoinbaseClicked() =
+        viewModelScope.launch {
+            val appId = BuildConfig.ZCASH_COINBASE_APP_ID
+
+            when {
+                appId.isEmpty() && isDebug ->
+                    coinbaseNavigationCommand.emit("https://www.coinbase.com") // fallback debug url
```

Comment:
Can we move these hardcoded urls to constants?

Reply:
no need to use constants here, especially for the formatted url below :) they are used only in 1 place